pkg/proxy/topology.go

```diff
@@ -137,29 +135,11 @@ func CategorizeEndpoints(endpoints []Endpoint, svcInfo ServicePort, nodeLabels m
 	return
 }
 
-// canUseTopology returns true if topology aware routing is enabled and properly
-// configured in this cluster. That is, it checks that:
-//   - The TopologyAwareHints or ServiceTrafficDistribution feature is enabled.
-//   - If ServiceTrafficDistribution feature gate is not enabled, then the
-//     hintsAnnotation should represent an enabled value.
+// canUseTopology returns true if all of the following is true:
 //   - The node's labels include "topology.kubernetes.io/zone".
 //   - All of the endpoints for this Service have a topology hint.
 //   - At least one endpoint for this Service is hinted for this node's zone.
-func canUseTopology(endpoints []Endpoint, svcInfo ServicePort, nodeLabels map[string]string) bool {
-	if !utilfeature.DefaultFeatureGate.Enabled(features.TopologyAwareHints) && !utilfeature.DefaultFeatureGate.Enabled(features.ServiceTrafficDistribution) {
-		return false
-	}
-
-	// Ignore value of hintsAnnotation if the ServiceTrafficDistribution feature
-	// gate is enabled.
-	if !utilfeature.DefaultFeatureGate.Enabled(features.ServiceTrafficDistribution) {
```

Comment:
@siyuanfoundation is my understanding correct and we should keep the feature gates in the code so we can emulate previous versions for 3 releases? if affirmative, I think we can be more explicit in https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/feature-gates.md#compatibility-versions

Reply:
oh, nevermind, this is kube-proxy, scratch my comment, I thought it was on the controller

Reply:
No worries, thanks! Yep this is on kube-proxy which is why I thought it's allowed (as has been explicitly called out for "Node components"). I have included the same in the commit message